subprojects/docs/src/docs/userguide/five_things.adoc

```diff
@@ -0,0 +1,126 @@
+
+== 2. The core model is based on a network of tasks
+
+Gradle models its builds as Directed Acyclic Graphs (DAGs) of tasks (units of work). What this means is that a build essentially configures a set of tasks and wires them together to create that DAG. Once the task graph has been created, Gradle determines which tasks need to be run in which order and then proceeds to execute them.
+
+This diagram shows two example task graphs, one abstract and the other concrete, with the dependencies between the tasks represented as arrows:
+
+.Two examples of a Gradle task graph
+image::task-dag-examples.png[Example task graphs]
+
+Almost any build process can be modeled as a graph of tasks in this way, which is one of the reasons why Gradle is so flexible.
+
+Tasks themselves consist of:
+
+ * Actions — pieces of work that do something, like copy files or compile source
+ * Inputs — values, files and directories that the actions use or operate on
+ * Outputs — files and directories that the actions modify or generate
+
+In fact, all of the above are optional depending on what the task needs to do. Some tasks — such as the <<base_plugin#sec:base_tasks,standard lifecycle tasks>> — don't even have any actions. They simply aggregate multiple tasks together as a convenience.
+
+NOTE: You choose which task to run. Save time by specifying the task that does what you need, but no more than that. If you just want to run the unit tests, choose the task that does that — typically `test`. If you want to package an application, most builds have an `assemble` task for that.
+
+One way to customize a build is to add extra steps to it. This is easy to do in Gradle: define your own tasks and insert them into the task graph via the <<tutorial_using_tasks#sec:task_dependencies,task dependency mechanism>>. You can have your tasks depend on any other task or configure existing tasks to depend on yours. The main restriction is that you cannot create any circular dependencies within the task graph.
+
+One last thing: Gradle's <<more_about_tasks#sec:up_to_date_checks,incremental build>> support is robust and reliable, so keep your builds running fast by avoiding the `clean` task unless you actually do want to perform a clean. 
+
+== 3. Gradle has distinct build phases
+
+It's important to understand that Gradle evaluates and executes build scripts in three phases:
+
+ 1. Initialization
+ 2. Configuration
+ 3. Execution
+
+In well-designed build scripts, the vast majority of what you see is evaluated during the configuration phase. Even so, some elements are likely to be evaluated during the execution phase, particularly the contents of `doLast {}` and `doFirst {}` blocks.
```

Comment:
If find this phrase problematic, as it implies that "as much as possible should happen at configuration time", the exact opposite of what we want. I think I know what you meant, but it's not clear to a newcomer.

Reply:
Correct. Will push soon...

Reply:
This doesn't really apply given new developments like lazy task configuration. Many of the blocks you see in a build scripts don't get executed at all in any given build. I would put the whole thing on its head and say: 

In order to get from startup to the actual task execution as quickly as possible, building a model of your projects and tasks needs to be fast. It is therefor important to defer any expensive computation (like resolving dependencies and walking file trees) to task execution time.